partiql-tests-data/eval-equiv/spec-tests.ion

```diff
@@ -0,0 +1,582 @@
+'section-4'::[
+    equiv_class::{
+        id: tuple_path_navigation,
+        statements: [
+            "{'a': 1, 'b': 2}.a",
+            "{'a': 1, 'b': 2}.\"a\""
+        ]
+    },
+    {
+        name: "equiv tuple path navigation",
+        statement: tuple_path_navigation,
+        assert: {
+            evalMode: [EvalModeCoerce, EvalModeError],
+            result: EvaluationSuccess,
+            output: 1
+        }
+    },
```

Comment:
Apologies if this has already been discussed and I've missed it: curious if we have considered the following format:

```bash
    {
        name: "equiv tuple path navigation",
        statements: [
            "{'a': 1, 'b': 2}.a",
            "{'a': 1, 'b': 2}.\"a\""
        ],
        assert: {
            evalMode: [EvalModeCoerce, EvalModeError],
            result: EvaluationSuccess,
            output: 1
        }
    },
```

Is that b/c we want to be able to refer to and share the equivalent classes? or we're considering the extending the usage of `EquivalenceClass` in future (considering it's mathematical semantics and utility)—I just want to double check we're not over-engineering/conceptualizing things.

Reply:
In https://github.com/partiql/partiql-tests/pull/26, after the initial equivalence test draft, I chose to model equivalence classes separately from test cases. Modeling them separately can allow us to decouple individual tests from equivalent statement tests. Modeling the equivalence class as a separate object with an identifiable symbol can also allow us to refer to this set of statements for other classes of tests (e.g. planner).

Reply:
I think your modeling is pretty succinct, but it ties the equivalent statements to the evaluation tests. Imo modeling it as a separate equivalence class allows for greater reuse (e.g. for planner tests) and doesn’t require too much additional syntax.

Reply:
I agree w/ re-usability of the current model; on the other hand, how do you see the current placement of the classes can help with sharing them across categories. For example, how can I re-use`tuple_path_navigation` for logical plan equivalence test? 

Side note: not directly related to this PR but I can also see some ambiguities (at least to me) which I think becomes more clear as we proceed. For example, should classes always be referred from statements, but I'm not expecting us to have a concrete answer for these.

Reply:
Perhaps the current layout of the classes isn't too clear on the reuse of the equivalence classes. Maybe it would be better to define all the equivalence classes in the top-level namespace? or top-level of file? But I could see reusing using the `tuple_path_navigation` using namespaces to separate the tests:

```
'section-4'::[
    equiv_class::{
        id: tuple_path_navigation,
        statements: [
            "{'a': 1, 'b': 2}.a",
            "{'a': 1, 'b': 2}.\"a\""
        ]
    },
    // some other equiv_class definitions
    evaluation_equivalence::[
        {
            name: "equiv tuple path navigation",
            statement: tuple_path_navigation,
            assert: {
                evalMode: [EvalModeCoerce, EvalModeError],
                result: EvaluationSuccess,
                output: 1
            }
        },
        // other evaluation_equivalence tests
    ],
    logical_plan_equivalence::[
        // logical plan equivalence tests that can refer to the namespace defined in the 'section-4' namespace
    ]
]

```

Reply:
I think we can save the discussion for another time whether these equivalence classes can be referred to outside their defined file (e.g. through importing a set of equivalence classes).

Reply:
Discussed this offline, putting the summary here for visibility:

Agree with not straying this PR but re-using the equiv classes only in a file (mostly for only one-off use) won't be as useful as being able to re-use them across categories which is the main driver for this design (E.g. re-using them for _logical plan_ )—we'll have a follow-on `TODO` referenced in this PR, and move on.

Reply:
Yeah, thanks for bringing up the point related to reuse of equivalence classes. Created #32 to track and added a TODO.